internal/pkg/checkin/bulk.go

```diff
@@ -240,6 +249,15 @@ func (bc *Bulk) flush(ctx context.Context) error {
 			if body, err = fields.Marshal(); err != nil {
 				return err
 			}
+
+			// If deleteAudit is set on the agent make sure the script is part of the request.
+			if pendingData.extra.deleteAudit {
+				updates = append(updates, bulk.MultiOp{
+					ID:    id,
+					Body:  deleteAuditAttributesScript,
+					Index: dl.FleetAgents,
```

Comment:
This is running the script for each agent individually? Would it be possible to do this in a single script?

Reply:
Yes, it runs the script for each agent ID in the batch.
My original approach had the script in an `update_by_query` request, but it kept getting version conflict errors (and there does not appear to be any retry on conflict option).

I can see if the update action is able to take both a `script` and `doc` as the payload, or we can change it so that the other attributes are set as part of the script so only one thing is ran (per agent)

Reply:
You can't provide both a script and doc for bulk update actions.

Reply:
I see. Looks like you updated it to just use a script to do both at the same time in the case those attribute needs to be removed. Looks good.